homeassistant/components/sensor/__init__.py

```diff
@@ -755,6 +754,28 @@ def state(self) -> Any:
                     f"but provides state {value}:{type(value)} resulting in '{err}'"
                 ) from err
 
+        # Sensors with device classes indicating a non-numeric value
+        # should not have a state class or unit of measurement
+        if device_class in {
+            SensorDeviceClass.DATE,
+            SensorDeviceClass.DURATION,
```

Comment:
Duration has a unit of measurement.

Reply:
Hmm I guess `duration` could have a state_class too in that case?

Reply:
Yeah, I think so.